src/amo/components/UserProfileEdit/index.js

```diff
@@ -77,9 +77,10 @@ export class UserProfileEditBase extends React.Component<Props, State> {
   }
 
   componentWillMount() {
-    const { currentUser, dispatch, errorHandler, username, user } = this.props;
+    const { dispatch, errorHandler, username, user } = this.props;
 
-    if (!currentUser) {
+    if (errorHandler.hasError()) {
```

Comment:
So, was this extra check not worth it? I guess, I could keep it somehow but.. do we really need it? 😕

Reply:
Sorry, but I'm totally confused now. Where do we stand with this? I'm not even sure which check you mean by the "extra check" above. Could you possibly summarize where we are with this?

Reply:
I don't think it was useful to have `if (!currentUser) { ... }` in the `componentWillMount()` method because I cannot realistically get an error. Dispatching the action below does not cause any issue.

Reply:
Ok, thanks @willdurand. As this was my only comment, this is now an r+. 👍